File: pingdart-php/src/Services/EmailService.php
<?php

namespace PingDart\SDK\Services;

use GuzzleHttp\Client;

class EmailService {
    private $http;

    public function __construct(Client $http) {
        $this->http = $http;
    }

    public function sendEmail(string $email, string $subject, string $text, array $smtpConfig = null) {
        $payload = [
            'email' => $email,
            'subject' => $subject,
            'text' => $text
        ];
        if ($smtpConfig) {
            $payload['smtpConfig'] = $smtpConfig;
        }

        try {
            $response = $this->http->post('email/send-email', [
                'json' => $payload
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Email sending error: " . $e->getMessage());
        }
    }

    public function bulkSend(array $data, array $smtpConfig = null) {
        $payload = ['data' => $data];
        if ($smtpConfig) {
            $payload['smtpConfig'] = $smtpConfig;
        }

        try {
            $response = $this->http->post('email/bulk-send', [
                'json' => $payload
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Bulk email sending error: " . $e->getMessage());
        }
    }
}

File: pingdart-python/pingdart_sdk/services/email_service.py
class EmailService:
    def __init__(self, http_client):
        self.http = http_client

    def send_email(self, email, subject, text, smtp_config=None):
        payload = {
            "email": email,
            "subject": subject,
            "text": text
        }
        if smtp_config:
            payload["smtpConfig"] = smtp_config
            
        try:
            response = self.http.post("/email/send-email", json=payload)
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Email sending error: {str(e)}")

    def bulk_send(self, data, smtp_config=None):
        payload = {
            "data": data
        }
        if smtp_config:
            payload["smtpConfig"] = smtp_config
            
        try:
            response = self.http.post("/email/bulk-send", json=payload)
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Bulk email sending error: {str(e)}")
